### PR TITLE
isort, doc8, and better RTD support

### PR DESCRIPTION
**Pull request by jmbowman**
* Now using isort to check compliance with the import order guidelines
* Now using doc8 to check style of reStructuredText files
* Run sphinx-apidoc in a custom Sphinx extension instead of a tox command, so it can be used on Read the Docs
* Renamed "pip-compile" make target to "upgrade", make it actually upgrade to new versions
* Updated some dependency versions
* Have requirements/dev.txt include quality check utilities for ease of running them individually
* Have requirements/doc.txt include base requirements (needed to avoid errors during sphinx-apidoc runs)

### CODE REVIEW

**Comment by jmbowman** [comment]
@macdiesel @doctoryes Up for reviewing this one also?

File: {{cookiecutter.repo_name}}/Makefile [hunk 1]
```diff
@@ -48,11 +48,11 @@ extract_translations: ## extract strings to be translated, outputting .mo files
...
 fake_translations: extract_translations dummy_translations compile_translations ## generate and compile dummy translation files
 
-pip-compile: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -q pip-tools
 	pip-compile --upgrade -o requirements/base.txt requirements/base.in
-	pip-compile --upgrade -o requirements/dev.txt requirements/dev.in
-	pip-compile --upgrade -o requirements/doc.txt requirements/doc.in
+	pip-compile --upgrade -o requirements/dev.txt requirements/dev.in requirements/quality.in
+	pip-compile --upgrade -o requirements/doc.txt requirements/base.in requirements/doc.in
```
Comment (doctoryes) [inline {{cookiecutter.repo_name}}/Makefile]: What's the purpose of this `Makefile`'s `upgrade` as opposed to the top-level `Makefile`'s `upgrade`?

File: {{cookiecutter.repo_name}}/requirements/dev.in [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Additional requirements for development of this application
 
 edx-lint                  # For updating pylintrc
--e git+https://github.com/edx/i18n-tools.git@v0.3.2#egg=i18n_tools==0.3.2  # For i18n_tool dummy
+edx-i18n-tools            # For i18n_tool dummy
```
Comment (doctoryes) [inline {{cookiecutter.repo_name}}/requirements/dev.in]: Hooray!

File: {{cookiecutter.repo_name}}/setup.cfg [hunk 3]
```diff
@@ -1,2 +1,10 @@
+[isort]
+line_length = 120
+known_edx =
+known_django = django
+known_djangoapp = model_utils
+known_first_party = {{ cookiecutter.app_name }}
+sections = FUTURE,STDLIB,THIRDPARTY,DJANGO,DJANGOAPP,EDX,FIRSTPARTY,LOCALFOLDER
```
Comment (doctoryes) [inline {{cookiecutter.repo_name}}/setup.cfg]: Neat.

File: {{cookiecutter.repo_name}}/Makefile (hunk 1 above)
Comment (jmbowman) [inline {{cookiecutter.repo_name}}/Makefile]: This is the Makefile that gets put into generated projects, where this target is used to update their dependencies.  The top-level one is for updating the dependencies of this repo (used in tests and quality checks of the template, etc.)

**Pull request merged**